clients/drcachesim/simulator/cache_fifo.cpp

```diff
@@ -62,6 +63,12 @@ cache_fifo_t::init(int associativity, int block_size, int total_size,
     return true;
 }
 
+void
+cache_fifo_t::invalidate_update(caching_device_block_t *block)
+{
+    // Leave counters unchanged for FIFO.
```

Comment:
Please augment the FIFO and LRU unit tests in cache_replacement_policy_unit_test.cpp to catch this bug without the fix and pass only with the fix for a regression test.

Reply:
The test now checks invalidations set the counters appropriately. I went and added tests for LFU as well. I also ran the test without the fixes and it breaks horribly (as expected).